File: STP/src/gestionAplicacion/Planeacion/Facturacion.java
package gestionAplicacion.Planeacion;
import gestionAplicacion.Personas.*;
import java.util.ArrayList;

public class Facturacion{
    private ArrayList<Usuario> usuario = new ArrayList<Usuario>();
    private ArrayList<Producto> productos = new ArrayList<Producto>();
    private double distancia;
    private double peso;
    private double precio;
    private double tiempo;

    public Facturacion(ArrayList<Usuario> usuario, ArrayList<Producto> productos, double distancia, double peso, double precio, double tiempo) {
        this.usuario = usuario;
        this.productos = productos;
        this.distancia = distancia;
        this.peso = peso;
        this.precio = precio;
        this.tiempo = tiempo;
    }
    public Facturacion() {}
    
    //GET Y SET
    
    public ArrayList<Usuario> getUsuario() {
        return usuario;
    }

    public ArrayList<Producto> getProductos() {
        return productos;
    }

    public double getDistancia() {
        return distancia;
    }

    public double getPeso() {
        return peso;
    }

    public double getPrecio() {
        return precio;
    }

    public double getTiempo() {
        return tiempo;
    }

    public void setUsuario(ArrayList<Usuario> usuario) {
        this.usuario = usuario;
    }

    public void setProductos(ArrayList<Producto> productos) {
        this.productos = productos;
    }

    public void setDistancia(double distancia) {
        this.distancia = distancia;
    }

    public void setPeso(double peso) {
        this.peso = peso;
    }

    public void setPrecio(double precio) {
        this.precio = precio;
    }

    public void setTiempo(double tiempo) {
        this.tiempo = tiempo;
    }
    
}

File: STP-Python/gestionAplicacion/Operatividad/Facturacion.py
class Facturacion():

    def __init__(self, *args):
        if len(args) == 6:
            self._pasajeros = args[0]
            self._productos = args[1]
            self._distancia = args[2]
            self._peso = args[3]
            self._precio = args[4]
            self._tiempo = args[5]
        elif len(args) == 0:
            self._pasajeros = []
            self._productos = []
            self._distancia = None
            self._peso = None
            self._precio = None
            self._tiempo = None
    
    def getPasajeros(self):
        return self._pasajeros
    
    def getProductos(self):
        return self._productos
    
    def getDistancia(self):
        return self._distancia
    
    def getPeso(self):
        return self._peso
    
    def getPrecio(self):
        return self._precio
    
    def getTiempo(self):
        return self._precio

    def setPasajeros(self, pasajeros):
        self._pasajeros = pasajeros
    
    def setProductos(self, productos):
        self._productos = productos

    def setDistancia(self, distancia):
        self._distancia = distancia
    
    def setPeso(self, peso):
        self._peso = peso

    def setPrecio(self, precio):
        self._precio = precio
    
    def setTiempo(self, tiempo):
        self._tiempo = tiempo